Realm/Tests/RealmTests.mm

```diff
@@ -1455,6 +1455,471 @@ - (void)testActiveVersionLimit {
     XCTAssertNotNil(error);
 }
 
+#ifdef REALM_ASYNC_WRITES
+#pragma mark - Async Transactions
+
+- (void)testAsyncTransactionShouldWrite {
+    RLMRealm *realm = RLMRealm.defaultRealm;
+    XCTestExpectation *asyncComplete = [self expectationWithDescription:@"async transaction complete"];
+
+    XCTAssertEqual(0, [StringObject allObjectsInRealm:realm].count);
+
+    [realm asyncTransactionWithBlock:^{
+        [realm createObject:StringObject.className withValue:@[@"string"]];
+    } onComplete:^{
+        StringObject *stringObject = [StringObject allObjectsInRealm:realm].firstObject;
+        XCTAssertEqualObjects(stringObject.stringCol, @"string");
+        [asyncComplete fulfill];
+    }];
+
+    [self waitForExpectationsWithTimeout:1.0 handler:nil];
+}
+
+- (void)testAsyncTransactionShouldWriteOnCommit {
+    RLMRealm *realm = RLMRealm.defaultRealm;
+    XCTestExpectation *asyncComplete = [self expectationWithDescription:@"async transaction complete"];
+
+    XCTAssertEqual(0, [StringObject allObjectsInRealm:realm].count);
+
+    [realm beginAsyncWriteTransaction:^{
+        [realm createObject:StringObject.className withValue:@[@"string"]];
+        [realm commitAsyncWriteTransaction:^{
+            StringObject *stringObject = [StringObject allObjectsInRealm:realm].firstObject;
+            XCTAssertEqualObjects(stringObject.stringCol, @"string");
```

Comment:
This test would pass even if it was moved out of this callback to before the call to `commitAsyncWriteTransaction:`, so it isn't actually testing much. A more effective way to test it would be to wrap it in `dispatchAsyncAndWait:` and obtain a new RLMRealm instance so that it's actually verifying that a new reader sees the committed data.